lib/pymedphys/tests/dicom/test_connect.py

```diff
@@ -187,3 +198,41 @@ def test_dicom_listener_send_conflicting_file(listener, test_dataset):
 
     # Clean up after ourselves
     shutil.rmtree(storage_path)
+
+
+@pytest.mark.pydicom
+def test_dicom_listener_cli(test_dataset):
+
+    scp_ae_title = "PYMEDPHYSTEST"
+
+    with process(
+        f"poetry run pymedphys dicom listen {TEST_PORT} -d {HERE} -a {scp_ae_title}",
+        shell=True,
+    ) as _:
+
+        # Send the data to the listener
+        ae = AE()
+        ae.add_requested_context(RTPlanStorage)
+        assoc = ae.associate("127.0.0.1", TEST_PORT, ae_title=scp_ae_title)
+
+        # Give the process a few seconds to start up
+        elapsed = 0
+        while not assoc.is_established:
+            time.sleep(0.5)
+            elapsed += 0.5
+            if elapsed >= 3:  # Break if still not connecting after 3 seconds
+                break
+
+            assoc = ae.associate("127.0.0.1", TEST_PORT, ae_title=scp_ae_title)
+        assert assoc.is_established
+        status = assoc.send_c_store(test_dataset)
+        assert status.Status == 0
+        assoc.release()
+
+    series_dir = HERE.joinpath(test_dataset.SeriesInstanceUID)
```

Comment:
With these sort of tests, we need to be able to support running multiple in parallel. For example, at times I want to spin up Python 3.6, 3.7, 3.8, and 3.9 and run the tests across all Python versions. As such, tests can't be creating and deleting files within a deterministic shared directory. Need to utilise tempdir creation here.